packages/react-instantsearch-dom/src/components/LinkList.js

```diff
@@ -35,7 +34,7 @@ export default class LinkList extends Component {
       <ul className={cx('list', !canRefine && 'list--noRefinement')}>
         {items.map(item => (
           <li
-            key={has(item, 'key') ? item.key : item.value}
+            key={item.key === undefined ? item.value : item.key}
```

Comment:
Why not use `hasOwnProperty` everywhere?

Reply:
Because you can only put undefined on purpose here with transformItems, which I can’t see having a reason. There’s already many more places we check for undefined indiscriminately. 

With translate I used hasOwnProperty because it’s a more dynamic piece of code, where passing undefined might be expected. 

However, I just realised that undefined doesn’t render nothing, meaning it can’t really have a reason to be used on purpose. Maybe that means we should revert the hasOwnProperty, and do use the default in that case. It can’t simply be a truthy check though

Reply:
IMO having the default check with `undefined` was fine even though it could be "breaking" (although the behavior would probably never happen).

We should be consistent across this file anyway. Strictly speaking, `hasOwnProperty` seems to better replace `lodash.has`, but it's fine using an `undefined` comparison here.

Reply:
I think you can even simplify the check to `Boolean(...)`. It was overkill to use the `has` function.